backend/sozisel/lib/sozisel_web/schema/resolvers/participant_resolvers.ex

```diff
@@ -186,4 +191,88 @@ defmodule SoziselWeb.Schema.Resolvers.ParticipantResolvers do
 
     verify_launched_event(ctx, launched_event_id, on_verify)
   end
+
+  def submit_whiteboard_result(
+        _parent,
+        %{
+          input: %{
+            launched_event_id: launched_event_id,
+            image: %Plug.Upload{} = image,
+            path: path,
+            text: text,
+            used_time: used_time
+          }
+        },
+        ctx
+      ) do
+    on_verified = fn %{
+                       launched_event: launched_event,
+                       session: session,
+                       participant: participant
+                     } ->
+      extension = Path.extname(image.filename)
+      filename = WhiteboardResult.generate_filename(launched_event.id, participant.id, extension)
+
+      event_result_attrs = %{
+        participant_id: participant.id,
+        launched_event_id: launched_event.id,
+        result_data: %{
+          metadata: %{},
+          path: filename,
+          text: text,
+          used_time: used_time
+        }
+      }
+
+      with %LaunchedEvent{event_id: event_id} <- Repo.get(LaunchedEvent, launched_event_id),
+           %Event{event_data: %event_data_module{} = event_data} <- Repo.get(Event, event_id),
+           :ok <- event_data_module.validate_result(event_data, event_result_attrs) do
+        Ecto.Multi.new()
+        |> Ecto.Multi.insert(
+          :whiteboard_event_result,
+          EventResult.create_changeset(%EventResult{}, event_result_attrs)
+        )
+        |> Ecto.Multi.run(:file_rename, fn _repo,
+                                           %{
```

Comment:
do the matching in function body so we do not get such huge indent.

Reply:
done